src/store/promoter/actions.ts

```diff
@@ -1,9 +1,11 @@
 import { createAction } from '@reduxjs/toolkit';
+import { combineRectsByRow } from '../../highlight/highlightUtil';
 import { SelectionItem } from './types';
 import { Shape } from '../../@types';
 
 export type SelectionArg = {
     location: number;
+    pageRect: DOMRect;
```

Comment:
Does this need to be stored?

Reply:
Yeah, it's needed. If you user selects text then scrolls page before click promotion button, the page x/y will not match with selection x/y. Have to get them and store them at the same moment.